src/mlpack/methods/hmm/hmm_impl.hpp

```diff
@@ -522,6 +522,78 @@ double HMM<Distribution>::LogLikelihood(const arma::mat& dataSeq) const
   return accu(logScales);
 }
 
+/**
+ * Compute the log of the scaling factor of the given emission probability
+ * at time t. To calculate the log-likelihood for the whole sequence,
+ * accumulate log scale over the entire sequence
+ */
+template<typename Distribution>
+double HMM<Distribution>::LogScaleEmissionProb(const arma::vec& emissionLogProb,
+                                        arma::vec& forwardLogProb) const
+{
+    double curLogScale;
+    if (forwardLogProb.empty())
+    {
+      // start of sequence or time t=0
+      forwardLogProb = ForwardAtT0(emissionLogProb, curLogScale);
+    }
+    else
+    {
+      forwardLogProb = ForwardAtTn(emissionLogProb, curLogScale,
+          forwardLogProb);
+    }
+
+    return curLogScale;
+}
+
+/**
+ * Compute the log-likelihood of the given emission probability up to time t
+ */
+template<typename Distribution>
+double HMM<Distribution>::LogLikelihoodEmissionProb(
+                                        const arma::vec& emissionLogProb,
+                                        double &logLikelihood,
+                                        arma::vec& forwardLogProb) const
+{
+    bool isStartOfSeq = forwardLogProb.empty();
+    double curLogScale = LogScaleEmissionProb(emissionLogProb, forwardLogProb);
+    logLikelihood = isStartOfSeq ? curLogScale : curLogScale + logLikelihood;
+    return logLikelihood;
+}
+
+/**
+  * Compute the log of the scaling factor of the given data at time t.
+  * To calculate the log-likelihood for the whole sequence, accumulate log
+  * scale over the entire sequence
+ */
+template<typename Distribution>
+double HMM<Distribution>::LogScale(const arma::vec &data,
+                                        arma::vec& forwardLogProb) const
+{
+    arma::vec emissionLogProb(logTransition.n_rows);
+
+    for (size_t state = 0; state < logTransition.n_rows; state++)
+    {
+          emissionLogProb(state) = emission[state].LogProbability(data);
+    }
+
+    return LogScaleEmissionProb(emissionLogProb, forwardLogProb);
```

Comment:
```suggestion
  arma::vec emissionLogProb(logTransition.n_rows);

  for (size_t state = 0; state < logTransition.n_rows; state++)
  {
    emissionLogProb(state) = emission[state].LogProbability(data);
  }

  return LogScaleEmissionProb(emissionLogProb, forwardLogProb);
```
Just some more style fixes. :)